api/dataprocessor.go

```diff
@@ -466,7 +467,7 @@ func (s *Server) getSeriesCachedStore(ctx *requestContext, until uint32) []chunk
 	// the request cannot completely be served from cache, it will require cassandra involvement
 	if !cacheRes.Complete {
 		if cacheRes.From != cacheRes.Until {
-			storeIterGens, err := s.BackendStore.Search(key, cacheRes.From, cacheRes.Until)
+			storeIterGens, err := s.BackendStore.Search(key, ctx.TTL, cacheRes.From, cacheRes.Until)
```

Comment:
I don't like introducing new dependencies. But I think before doing that I should wait for this https://github.com/raintank/metrictank/pull/509 to get merged, because of the modifications it's making in `alignRequests()` which would be useful here.